custom_components/tuya_local/devices/atorch_s1wp.yaml

```diff
@@ -0,0 +1,575 @@
+name: Energy monitoring plug with display
+products:
+  - id: sqrf2g1amfutn4co
+    name: Atorch S1WP
+primary_entity:
+  entity: switch
+  class: outlet
+  dps:
+    - id: 1
+      type: boolean
+      name: switch
+secondary_entities:
+  - entity: number
+    category: config
+    name: Countdown
+    icon: "mdi:timer"
+    dps:
+      - id: 9
+        type: integer
+        name: value
+        unit: min
+        range:
+          min: 0
+          max: 360000
+        mapping:
+          - scale: 60
+            step: 60
+          - scale: 1000
+  - entity: sensor
+    class: current
+    name: Current
+    dps:
+      - id: 18
+        name: sensor
+        type: integer
+        class: measurement
+        unit: mA
+        mapping:
+          - scale: 1000
+  - entity: sensor
+    class: power
+    name: Power
+    dps:
+      - id: 19
+        name: sensor
+        type: integer
+        class: measurement
+        unit: W
+        mapping:
+          - scale: 100
+  - entity: sensor
+    class: voltage
+    name: Voltage
+    dps:
+      - id: 20
+        name: sensor
+        type: integer
+        class: measurement
+        unit: V
+        mapping:
+          - scale: 100
+  - entity: number
+    category: config
+    name: Electricity price
+    icon: "mdi:cash-multiple"
+    dps:
+      - id: 101
+        type: integer
+        name: value
+        range:
+          min: 0
+          max: 99999
+        mapping:
+          - scale: 100
+            step: 1
+  - entity: sensor
+    name: Total Cost
+    dps:
+      - id: 102
+        name: sensor
+        type: integer
+        mapping:
+          - scale: 1000
+  - entity: number
+    category: config
+    name: Overvoltage value
+    icon: "mdi:flash-triangle"
+    dps:
+      - id: 104
+        type: integer
+        name: value
+        unit: V
+        range:
+          min: 1
+          max: 2750
+        mapping:
+          - scale: 10
+            step: 1
+  - entity: number
+    category: config
+    name: Overcurrent value
+    icon: "mdi:flash-triangle"
+    dps:
+      - id: 105
+        type: integer
+        name: value
+        unit: A
+        range:
+          min: 1
+          max: 2000
+        mapping:
+          - scale: 100
+            step: 1
+  - entity: number
+    category: config
+    name: Overpower value
+    icon: "mdi:flash-triangle"
+    dps:
+      - id: 106
+        type: integer
+        name: value
+        unit: W
+        range:
+          min: 1
+          max: 4500
+        mapping:
+          - scale: 1
```

Comment:
`scale: 1` is also not needed, so this whole mapping block can be removed.

Reply:
i was just annoyed by debug log recommending to add mapping for the dp